dirwatcher.js

```diff
@@ -0,0 +1,72 @@
+import EventEmitter from 'events';
+import {readdir} from 'fs';
+
+function filterByCSVExtension(filename) {
+  return filename.endsWith('.csv');
+}
+
+export default class DirWatcher extends EventEmitter {
+  constructor() {
+    super();
+  }
+
+  postError(error) {
+    this.emit(`error: ${error}`);
+  }
+
+  postMessageChange(data) {
+    this.emit('dirwatcher:changed', data);
+  }
+
+  postMessageCreated({path, filename}) {
+    this.postMessageChange({path, filename, message: `${filename} was created`});
+  }
+
+  postMessageRemoved(filename) {
+    this.postMessageChange(`${filename} was removed`);
+  }
+
+  watch(path, delay) {
+    console.log(`watching files in ${path}…`);
+
+    readdir(path, (error, files) => {
+      if (error) {
+        this.postError(error);
+        return;
+      }
+
+      const prevFilesState = files.filter(filterByCSVExtension);
+
+      for (const filename of prevFilesState) {
+        this.postMessageCreated({path, filename});
+      }
+
+      const interval = setInterval(() => {
+        readdir(path, (error, files) => {
+          const curFilesState = files.filter(filterByCSVExtension);;
+          if (error) {
+            clearInterval(interval);
+            this.postError(error);
+            return;
+          }
+
+          if (curFilesState.length >= prevFilesState.length) {
```

Comment:
Your cases does not cover the case when a file was renamed, i.e. collection length was not changed, but the handler is executed.